backend/src/db/migrations/20250502230521-migrate-to-s3.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+
+import { open } from 'node:fs/promises'
+import path from 'node:path'
+
+import { S3Client, ObjectCannedACL } from '@aws-sdk/client-s3'
+import { Upload } from '@aws-sdk/lib-storage'
+import { lookup } from 'mime-types'
+
+import CONFIG from '@config/index'
+import { getDriver } from '@db/neo4j'
+
+export const description =
+  'Upload all image files to a S3 compatible object storage in order to reduce load on our backend.'
+
+export async function up(_next) {
+  if (CONFIG.NODE_ENV === 'test') {
+    // Let's skip this migration for simplicity.
+    // There is nothing to migrate in test environment and setting up the S3 services seems overkill.
```

Comment:
i agree that it is not necessary for test environment to migrate stuff, but since the s3 will be the only option for image handling we need to set it up as well. The E2E tests for example test for correct behaviour if I am correct.

Reply:
OK! I can for sure setup a test S3 service - it might slow down CI a bit.

Reply:
You will find an appropriate solution - just have a look at https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/master/cypress/e2e/UserProfile.Avatar.feature